integreat_cms/core/signals/hix_signals.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+from django.conf import settings
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from ...cms.models import PageTranslation
+from ...cms.views.utils.hix import lookup_hix_score
+
+logger = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=PageTranslation)
+def page_translation_save_handler(instance, **kwargs):
+    r"""
+    Calculates a hix store for a page translation after saving
+
+    :param instance: The page translation that gets saved
+    :type instance: ~integreat_cms.cms.models.pages.page_translation.PageTranslation
+
+    :param \**kwargs: The supplied keyword arguments
+    :type \**kwargs: dict
+    """
+    if instance.hix_score is not None:
+        return
+
+    if not hix_enabled(instance):
+        return
```

Comment:
```suggestion
    if instance.hix_score or not hix_enabled(instance):
        return
```
Unless I'm mistaken here, is there a special reason to compare with `None`?

I'm also not really clear on why we need to ckeck if the score exists already, is it saved multiple times otherwise?

Reply:
If we don't check whether the score was already calculated, the server ends up in an infinite loop, because calculating and saving the hix score causes the save signal handler to run again. Without the check it would calculate the hix score again, save it and then run again due to the save.

I just compared to None because I find it easier understandable. Do we have a guideline whether to use `is not None` or the implicit version? It seems like we use both variants in the codebase

Reply:
> If we don't check whether the score was already calculated, the server ends up in an infinite loop, because calculating and saving the hix score causes the save signal handler to run again. Without the check it would calculate the hix score again, save it and then run again due to the save.

Ah, gotcha. That makes sense.


> I just compared to None because I find it easier understandable. Do we have a guideline whether to use `is not None` or the implicit version? It seems like we use both variants in the codebase

Don't know if we have guidelines, and there are cases when `if A` =/= `if A is not None`, but I did not see that here.

TBH I feel like with your explanation about score re-calculation it makes even more intuitive sense to say "if the score exists, return" 😅 But ultimately it's just a style preference.

Reply:
@david-venhoff In my opinion, we should use `is not None` only if we want to differentiate between `False`, `0`, `""` (or other falsy values) and `None`. In all cases where it doesn't make a difference, I'd prefer `not x`.
Especially when handling with the Django ORM, we can safely assume that any object that is not `None` automatically evaluates to any truthy value, and I would also apply this pattern on our own custom methods.